### PR TITLE
Contructed inventory add edit form

### PR DESCRIPTION
**Pull request by MilanPospisil**
Create and Edit constructed inventory. Enhance the InventoryForm, which is used for all 3 types - inventory, smart and constructed.

Also:
- created Multi selector for AWX resources
- added QueryParams and LabelHelp to both single and multi AWX resources
- disabled Inventory form create and edit submit on enter, so labels can be added by typing and pressing enter
(without that, the user will type and possibly press enter and he dont want the form to be submitted).
- added error handling for edit form data loading and loading spinner     
- component and e2e tests will come in follow ups PRs

image

### CODE REVIEW

File: frontend/awx/common/PageFormMultiSelectAwxResource.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      title={props.title}
+      onSelect={props.onSelect}
+      toolbarFilters={props.toolbarFilters ?? []}
+      tableColumns={props.tableColumns}
+      view={view}
+    />
+  );
+}
+
+export function constructQueryParams(queryParams: QueryParams) {
```
Comment (prat98) [inline frontend/awx/common/PageFormMultiSelectAwxResource.tsx:161]: Do we want to put this function in a separate file maybe under utils?

Comment (MilanPospisil) [reply frontend/awx/common/PageFormMultiSelectAwxResource.tsx:161]: I will look if there is any usage of this functions in fw.

**Comment by prat98** [comment]
Found one more thing. Need to change `PageFormCodeEditor` label from `Variables` to `Source vars` and add a help text for that section of the form. In addition, this input should be a required field and needs to have validation on it to ensure a `plugin` parameter is provided in the source vars.
Screenshot 2024-03-28 at 11 31 20 AM

**Review by prat98 — APPROVED** [review]
LGTM! 🚀

**Pull request merged**